### PR TITLE
[IMP] *: improve domain selector edition, dom, and style

### PR DESCRIPTION
**Pull request by Polymorphe57**
### Main goals

- Make the domain selector be able to display all domains accepted by the class Domain (domain.js)
- Improve the edition/selection of values, e.g. for the operators in/not in
- Improve the style/dom

### Changes/Specification

**Edition of conditions**

We call _condition_ a triple (_path_, _operator_, _value_). Each component can be edited via a dedicated _editor_.

**Path editor** 

- a _valid_ path (a string or 0/1) is accepted by the path editor and edited via the model field selector.
- an _invalid_ path is displayed with an error message "Invalid field chain" and a clear button. The button clear
  reset the path to a valid default path (e.g. ID).

**Operator editor**

- displays a selection of operators determined by the path field type (if any)
- a _valid_ operator (see TERM_OPERATORS in expression.py) is added temporarily to the selection if missing
- an __invalid_ operator is displayed with an error message "Invalid operator" and a clear button. The button clear
  reset the operator to a valid default operator (determined by the path field type)

**Value editors**

- the value editor used is determined by the path field type (if any) and the operator
- each value editor has a set of supported value types
- a value not supported is displayed with an error message, e.g. "Value not supported", and a clear button. The clear  
  button reset the value to a default value accepted by the editor
- (_basic editor_) In general, a supported value is edited via an input or a select/autocomplete dropdown when 
  appropriate or a datetime picker.
- editor for _in/not in_ operators: the current value is displayed as a list of tags (removable) and a value can be added via 
  the suitable basic editor.
- editor for _between_: two suitable basic editors are displayed

**DOM/Style**

- The dom has been refactored and now contains less elements/custom classes
- The stylesheet has been almost emptied
- Bootstrap is used intensively

Task IDs: 3507123, 3508940

### CODE REVIEW

File: addons/web/static/src/core/domain_selector/domain_selector.xml [hunk 1]
```diff
@@ -2,85 +2,94 @@
 <templates xml:space="preserve">
 
     <t t-name="web._DomainSelector">
```
Comment (mcm-odoo) [inline addons/web/static/src/core/domain_selector/domain_selector.xml:4]: rename this template, it was like this because of the duplication with legacy

Comment (Polymorphe57) [reply addons/web/static/src/core/domain_selector/domain_selector.xml]: done

File: addons/web/static/src/core/model_field_selector/model_field_selector.xml [hunk 2]
```diff
@@ -2,13 +2,13 @@
 <templates xml:space="preserve">
 
     <t t-name="web._ModelFieldSelector">
```
Comment (mcm-odoo) [inline addons/web/static/src/core/model_field_selector/model_field_selector.xml:4]:
```suggestion
    <t t-name="web.ModelFieldSelector">
```

File: addons/web/static/src/core/domain_selector/domain_selector_operator_editor.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    if (typeof operator === "string" && operator in OPERATOR_DESCRIPTIONS) {
+        if (negate && operator in TERM_OPERATORS_NEGATION) {
+            return OPERATOR_DESCRIPTIONS[TERM_OPERATORS_NEGATION[operator]];
+        }
+        label = OPERATOR_DESCRIPTIONS[operator];
+    } else {
+        label = formatValue(operator);
+    }
+    if (negate) {
+        return sprintf(`not %s`, label);
```
Comment (mcm-odoo) [inline addons/web/static/src/core/domain_selector/domain_selector_operator_editor.js:142]:
```suggestion
        return `not ${label}`;
```

Comment (Polymorphe57) [reply addons/web/static/src/core/domain_selector/domain_selector_operator_editor.js:142]: remove sprintf everywhere and use _t

Comment (Polymorphe57) [reply addons/web/static/src/core/domain_selector/domain_selector_operator_editor.js:142]: import * as dsHelpers from ... (shorter)

File: addons/web/static/src/core/model_field_selector/model_field_selector.xml (hunk 2 above)
Comment (Polymorphe57) [reply addons/web/static/src/core/model_field_selector/model_field_selector.xml:4]: elsewhere

File: addons/web/static/src/core/domain_selector/domain_selector_operator_editor.js (hunk 3 above)
Comment (Polymorphe57) [reply addons/web/static/src/core/domain_selector/domain_selector_operator_editor.js:142]: done

**Comment by mcm-odoo** [comment]
@robodoo r+

**Comment by mcm-odoo** [comment]
@robodoo r+

**Comment by Polymorphe57** [comment]
robodoo r-

**Comment by mcm-odoo** [comment]
@robodoo r+

**Pull request closed without merging**